src/Razor/src/Microsoft.AspNetCore.Razor.VSCode.BlazorWasmCompanionExtension/src/extension.ts

```diff
@@ -0,0 +1,72 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+
+import * as vscode from 'vscode';
+import { LocalDebugProxyManager } from './LocalDebugProxyManager';
+import { spawn } from 'child_process';
+
+let launchDebugProxy: vscode.Disposable;
+let killDebugProxy: vscode.Disposable;
+
+export function activate(context: vscode.ExtensionContext) {
+    const outputChannel = vscode.window.createOutputChannel("Blazor WASM Debug Proxy");
+    const pidsByUrl = new Map<string, number>();
+
+    launchDebugProxy = vscode.commands.registerCommand('ms-blazorwasm-companion.launchDebugProxy', async (version = "5.0.0") => {
+        try {
+            outputChannel.appendLine(`Launching proxy version Blazor WASM ${version}...`);
+            const localDebugProxyManager = new LocalDebugProxyManager();
+
+            const debuggingPort = await LocalDebugProxyManager.getAvailablePort(9222);
+            const debuggingHost = `http://localhost:${debuggingPort}`;
+
+            const debugProxyLocalDirectory = await localDebugProxyManager.getDebugProxyLocalNugetPath(version);
+            const debugProxyLocalPath = `${debugProxyLocalDirectory}/tools/BlazorDebugProxy/BrowserDebugHost.dll`;
+            outputChannel.appendLine(`Launching debugging proxy from ${debugProxyLocalPath}`);
+            const spawnedProxy = spawn('/usr/local/share/dotnet/dotnet',
+                [debugProxyLocalPath , '--DevToolsUrl', debuggingHost],
+                { detached: process.platform !== 'win32' });
+
+            for await (const output of spawnedProxy.stdout) {
+                outputChannel.appendLine(output);
+                const matchExpr = "Now listening on: (?<url>.*)";
+                const found = `${output}`.match(matchExpr);
+                const url = found?.groups?.url;
+                if (url) {
+                    outputChannel.appendLine(`Debugging proxy is running at: ${url}`);
```

Comment:
Correct. The stdout here is basically a line-by-line stream of the result from running the debugging proxy. Looks something like this:

```
Hosting environment: Production
Content root path: /Users/captainsafia/.blazor-wasm-debug-proxy-local/extracted/Microsoft.AspNetCore.Components.WebAssembly.DevServer.5.0.0/tools/BlazorDebugProxy
Now listening on: http://127.0.0.1:51134
Application started. Press Ctrl+C to shut down.
```

Reply:
Ya this was confusion on my part from the "chunking" behavior you mentioned in a previous post

Reply:
Given the current API/usage what happens if there's a change in behavior from dotnet or something else that results in `url` never being found? Wont the await foreach above infinite loop and effectively hang the world? Makes me think it may be something we want to "timeout" on and notify the user

Reply:
>  Given the current API/usage what happens if there's a change in behavior from dotnet or something else that results in url never being found?

It depends on what the change in behavior is. The `stdout.stream` will break the for-loop when that stream is closed.

If that doesn't happen, then yes, infinite loop. I added a trial based timeout here.

> Given the current API/usage what happens if there's a change in behavior from dotnet or something else that results in url never being found?

The real answer to this is we should stay on top of it and ship a fix out immediately. However, since this semantic is heavily baked into a lot of different things, I hope the chances of it happening are slim.